File: backend/src/main/java/com/example/stamp_app/entity/Sensor.java
package com.example.stamp_app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.validation.constraints.NotNull;
import org.hibernate.validator.constraints.Length;

import java.util.List;

@Entity
public class Sensor {

    @Id
    @Length(max = 3, min = 1)
    private Long id;

    @Column
    @NotNull
    private String sensorName;

    @Column
    private String vendorName;

    @Column
    private String identificationCode;

    @OneToMany(mappedBy = "sensor")
    private List<Sdi12Data> sdi12Data;

}

File: backend/src/main/java/com/example/stamp_app/entity/Sensor.kt
package com.example.stamp_app.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.validation.constraints.NotNull
import org.hibernate.validator.constraints.Length

@Entity
data class Sensor (
    @Id
    val id: @Length(max = 3, min = 1) Long? = null,

    @Column
    val sensorName: @NotNull String? = null,

    @Column
    val vendorName: String? = null,

    @Column
    val identificationCode: String? = null,

    @OneToMany(mappedBy = "sensor")
    val sdi12Data: List<Sdi12Data>? = null
)
